spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1396,6 +1396,17 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde {
           val optExpr = scalarExprToProto("atan2", leftExpr, rightExpr)
           optExprWithInfo(optExpr, expr, left, right)
 
+        case e @ Unhex(child, failOnError) =>
```

Comment:
https://github.com/apache/spark/blob/59d5946cfd377e9203ccf572deb34f87fab7510c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala#L1152
vs 
https://github.com/apache/spark/blob/45ba9224602eb18fe45e339cbb8cf2e8a4924f0b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala#L1125

Reply:
We need to use a shim approach. We currently have one folder with source for spark-3.x. We should add additional folders as needed e.g. spark-3.3.x / spark-3.4.x with any code that is specific to those versions.

I am happy to help with this.

Reply:
I think we can handle this using existing shim, something like
```
case e @ Unhex =>
e.child...
getFailOnError(e) ...
```

Reply:
I was wrong, actually `getFailOnError` needs to be modified for this approach as `unhex` is `unaryExpression`

Reply:
@andygrove Would you mind having a look at the updated PR and let me know what you think of the shim implemntation? The thought was to setup a major and minor shim to keep the existing shim for all of spark 3 then apply the more granular difference in the minor. It gets a little weird in that I'm applying 3.3's shim to 3.2.

@kazuyukitanimura Thanks for looking!

Reply:
Thanks @tshauck I will review this tomorrow.

Reply:
I think we can have duplicated shim for 3.3 and 3.2